cdap-app-fabric/src/main/java/io/cdap/cdap/config/PreferencesService.java

```diff
@@ -53,35 +59,42 @@
  * in transaction in each method
  */
 public class PreferencesService {
-
+  private final AuthorizationEnforcer authorizationEnforcer;
+  private final AuthenticationContext authenticationContext;
   private final AdminEventPublisher adminEventPublisher;
   private final TransactionRunner transactionRunner;
 
   @Inject
-  public PreferencesService(MessagingService messagingService,
+  public PreferencesService(AuthorizationEnforcer authorizationEnforcer,
+                            AuthenticationContext authenticationContext,
+                            MessagingService messagingService,
                             CConfiguration cConf, TransactionRunner transactionRunner) {
+    this.authorizationEnforcer = authorizationEnforcer;
+    this.authenticationContext = authenticationContext;
     MultiThreadMessagingContext messagingContext = new MultiThreadMessagingContext(messagingService);
     this.adminEventPublisher = new AdminEventPublisher(cConf, messagingContext);
     this.transactionRunner = transactionRunner;
   }
 
-  private PreferencesDetail get(EntityId entityId) {
+  private PreferencesDetail get(EntityId entityId) throws Exception {
+    AuthorizationUtil.ensureAccess(entityId, authorizationEnforcer, authenticationContext.getPrincipal());
```

Comment:
same here

Reply:
As discussed, auth extension will return if it is an instance id